dashboard/app/controllers/api/v1/sections_students_controller.rb

```diff
@@ -10,4 +12,25 @@ def index
       )
     end)
   end
+
+  # PATCH /sections/<section_id>/student/<id>/update
+  def update
+    student = @section.students.find_by_id(params[:id])
+
+    if student.update(student_params)
+      render json: student.summarize
+    else
+      render json: {errors: student.errors.full_messages}, status: :bad_request
+    end
+  end
+
+  def student_params
+    allowed_params = [
+      :age,
+      :gender,
+      :name,
+      :sharing_disabled,
+    ]
+    params.permit(*allowed_params)
```

Comment:
note, these can be combined. I think the example I sent you earlier had them separated because it had some branching logic.

```ruby
params.permit(
  :age,
  :gender,
  :name,
  :sharing_disabled
)
```

Also Rails [strong parameters](http://api.rubyonrails.org/classes/ActionController/StrongParameters.html) are usually encapsulated in a container named after the model, so ideally to fit convention, this should be (regardless of whether or not you define `allowed_params` separately):

```ruby
params.require(:student).permit(
  :age,
  :gender,
  :name,
  :sharing_disabled
)
```
The main advantage of doing it this way is extensibility. Everything passed inside a `student` key in the data payload gets sent to the student model, and you have the option of adding additional top level parameters in the future without breaking the existing API.

Reply:
Whenever I added `.require(:student)` or `.require(:user)` the API returns a 404. I'm not sure what the model is because they are students for a section.

Reply:
The relevant model is `school` because that is the one being updated. The section is just a scope, a container for the school (for auth purposes). 

You're getting a 404 because the data payload needs to be updated too. [Params.require](http://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-require) will raise an error when the required param is missing, which Rails translates to a 404. It's much like if you left out your `id` parameter, which is implicitly required from the route definition. To get it to work, encapsulate the school params inside a school container:

```
{
  school: { // this part is required
    age: ..., // these are permitted, i.e. optional
    // etc
  }
}
```